class.jetpack-onboarding-end-points.php

```diff
@@ -539,6 +545,41 @@ static function install_theme() {
 		wp_send_json_success( $theme_id );
 	}
 
+	static function install_plugin( $slug ) {
+		if ( is_multisite() && ! current_user_can( 'manage_network' ) ) {
+			return new WP_Error( 'not_allowed', 'You are not allowed to install plugins on this site.' );
+		}
+		include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+		include_once ABSPATH . 'wp-admin/includes/file.php';
+		$upgrader  = new Plugin_Upgrader( new Automatic_Upgrader_Skin() );
+		$zip_url = "https://downloads.wordpress.org/plugin/{$slug}.latest-stable.zip";
+		$result = $upgrader->install( $zip_url );
+		if ( is_wp_error( $result ) || ! $result ) {
+			return new WP_Error( 'install_error', 'Could not install plugin ' . $slug );
+		}
+		return true;
+	}
+
+	static function install_woocommerce() {
+		check_ajax_referer( self::AJAX_NONCE, 'nonce' );
+		$result = true;
+		if ( ! array_key_exists( self::WOOCOMMERCE_ID, get_plugins() ) ) {
+			$installed = self::install_plugin( self::WOOCOMMERCE_SLUG );
+			if ( is_wp_error( $installed ) ) {
+				$result = $installed;
+			} else {
+				$result = activate_plugin( self::WOOCOMMERCE_ID );
+			}
+		} else if ( ! is_plugin_active( self::WOOCOMMERCE_ID ) ) {
+			$result = activate_plugin( self::WOOCOMMERCE_ID );
```

Comment:
Prior to activating a plugin, we should check if the user has the right capabilities to do so: `current_user_can( 'activate_plugins' )`